src/js/jsx/search/SearchBar.jsx

```diff
@@ -142,6 +193,36 @@ define(function (require, exports, module) {
         },
 
         /**
+         * Get the layer type
+         *
+         * @private
+         * @param {Layer} layer
+         * @return {Array.<string>}
+         */
+        _getLayerType: function (layer) {
+            var layerType = ["layer"];
+            _.forEach(Object.keys(layer.layerKinds), function (kind) {
```

Comment:
The `layerKinds` property refers to the same object on every layer which describes ALL the available layer kinds. It's a convenience property that's kind of confusing. The `kind` property on layer models describes their individual kind.